packages/gestalt/src/Dropdown.js

```diff
@@ -0,0 +1,247 @@
+// @flow strict
+import React, { useState, useRef, type Node } from 'react';
+import PropTypes from 'prop-types';
+import Box from './Box.js';
+import Flyout from './Flyout.js';
+import Layer from './Layer.js';
+import DropdownItem from './DropdownItem.js';
+import DropdownSection from './DropdownSection.js';
+import DropdownContext from './DropdownContextProvider.js';
+import handleContainerScrolling from './utils/keyboardNavigation.js';
+
+type OptionObject = {|
+  label: string,
+  value: string,
+  subtext?: string,
+|};
+type Props = {|
+  anchor?: ?HTMLElement,
+  children: Node,
+  headerContent?: Node,
+  idealDirection?: 'up' | 'right' | 'down' | 'left',
+  onDismiss: () => void,
+  onSelect?: ({|
+    event: SyntheticKeyboardEvent<HTMLElement> | {| keyCode: number |},
+    item: ?OptionObject,
+  |}) => void,
+|};
+
+export default function Dropdown({
+  anchor,
+  children,
+  headerContent,
+  idealDirection = 'down',
+  onDismiss,
+  onSelect,
+}: Props): Node {
+  const getFlattenedChildren = () => {
+    const dropdownChildrenArray = React.Children.toArray(children);
+
+    const items = dropdownChildrenArray.map((child) => {
+      if (
+        child.props.children &&
+        child.type.displayName === 'DropdownSection'
+      ) {
+        return child.props.children;
+      }
+      if (child.type.displayName === 'DropdownItem') {
+        return child;
+      }
+      return null;
+    });
+
+    return items.flat();
+  };
+
+  const availableOptions = getFlattenedChildren();
```

Comment:
`const availableOptions=Children.toArray(children).reduceChildren.toArray(children).reduce...`

I wonder if it's worth adding a tiny comment explaining what this is doing... maybe not.

Reply:
The comment seems unnecessary as long as the code is still organized as the getFlattenedChildren function. Variable naming ftw! 🙂